woocommerce/woocommerce
# [Experimental] Add default product attribute pattern to Product Filters template part

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR:
- Updates the `ProductFilterAttribute::get_default_attribute()` method with a more explicit name and fixes its return type and value.
- Adds a new hidden pattern named `woocommerce/default-product-attribute` that dynamically shows the expected default product attribute for each store.
- Updates the `product-filters.html` template part to use the new pattern.

Closes #49978.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with the `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
1. Reset any customization made to the `Product Filters` template part.
2. Add the template part to the Product Catalog template.
3. In the Editor, see the expected attribute is chosen, with the correct title.
4. On the front end, visit the shop page, see the expected attribute and correct is shown, which matches the editor preview.
5. Repeat the steps above on another site with another set of attributes, or make changes to an attribute that make it meet the criteria for the default one. See it reflects on the front end as well as the editor.

What are the criteria for the default product attribute:

> We pre-select an attribute with the most terms (but no more than 30).

In practice, we select the first attribute that has several terms that are closest to 30. The store may have all attributes having more than 30 terms.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
